### ISSUE TITLE
Programatically generate blocks/scripts

### ISSUE THREAD

**Issue by sp00x** [dbaldwin/DroneBlocks-Web#19]
Is or would it possible to somehow externally generate/manipulate the scripts that DroneBlocks uses, so that I could e.g. write a small script that based on a calculated trajectory and set of actions can write the script / blocks for how I want a mission performed?

E.g.

Take off

Change altitude to A ft
Change gimbal tilt to B
Move forward C ft
<other parameters>
Take photo

Change altitude to D ft
Change gimbal tilt to E
Move forward F ft
<other parameters>
Take photo

etc..

My expressions for calculating the flight path seems to be a bit more than I would try to add using the variable language available.

**Comment by kbquist** [comment]
That's an interesting use case/mission. The altitude and forward inputs ought to be easily handled by variables.  For the gimbal pitch, what is the formula you would use?  Are you trying to keep the gimbal pointed at the same spot on the ground from all these waypoints?

**Comment by dbaldwin** [comment]
@sp00x If you can provide a more explicit example we'd love to discuss internally. We've talked about allowing external scripting in a future version, but perhaps we can get there quicker with some  feedback from users.

**Comment by sp00x** [comment]
Yes, the idea I had was a batching of photo jobs like for creating "flatland" and other more complex stitching concepts. It involves some maths that I would not want to do with "blocks" and rather write the expressions for myself. (I'm doing a small javascript test page - I'll post a link once I'm done..)

**Comment by dbaldwin** [comment]
Great. Yes please share that when you have a proof of concept. Thanks.

**Comment by dbaldwin** [comment]
@sp00x why don't you send me an email and I'll get you on the beta which will allow you to "script" instead of "block". My email is db at unmannedairlines dot com. Look forward to hearing from you.

**Comment by sp00x** [comment]
@dbaldwin Hi, sorry - I got distracted here by various unrelated stuff, and then I started trying to map the flight path using three.js and to simulate the viewport/photos that would be taken from the different waypoints.. and then things spiraled out of scope again as they tend to 8)

I'll drop you an email!
